src/traversals/eulerian.jl

```diff
@@ -0,0 +1,93 @@
+# Adapted from SimpleGraphs.jl [Copyright (c) 2014, Ed Scheinerman]:
+#    https://github.com/scheinerman/SimpleGraphs.jl/blob/master/src/simple_euler.jl
+# Reproduced under the MIT Expat License.
+
+"""
+    eulerian(g::AbstractSimpleGraph{T}[, u::T]) --> T[]
+
+Returns a [Eulerian trail or cycle](https://en.wikipedia.org/wiki/Eulerian_path) through an
+undirected graph `g`, starting at `u`, returning a vector listing the vertices of `g` in the
+order that they are traversed. If no such trail or cycle exists, throws an error.
+
+## Optional arguments
+- If `u` is omitted, a Eulerian trail or cycle is computed with `u = first(vertices(g))`.
+"""
+function eulerian(g::AbstractSimpleGraph{T}, u::T=first(vertices(g))) where {T}
```

Comment:
Why not an `AbstractGraph`?

Reply:
Ah, this was because I was not aware of the `IsDirected` trait when I wrote this. Changed to `AbstractGraph` now.